app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -279,14 +279,23 @@ class __ChatInputState extends ConsumerState<_ChatInput> {
   }
 
   void handleEmojiSelected(Category? category, Emoji emoji) {
+    String suffixText = '';
+
+    // Get the left text of cursor
+    String prefixText = '';
     // Get cursor current position
     var cursorPos = textController.selection.base.offset;
+    if (cursorPos >= 0) {
+      // can be -1 on empty and never accessed
 
-    // Right text of cursor position
-    String suffixText = textController.text.substring(cursorPos);
+      // Right text of cursor position
+      suffixText = textController.text.substring(cursorPos);
 
-    // Get the left text of cursor
-    String prefixText = textController.text.substring(0, cursorPos);
+      // Get the left text of cursor
+      prefixText = textController.text.substring(0, cursorPos);
+    } else {
+      cursorPos = 0;
+    }
```

Comment:
May I know reason for this condition?

Reply:
well, if the field never had focus, the cursorPos is `-1`. which then fails when you do `.substring(-1)` (new line 292) without executing the remaining code.

With this fix, this now sets the cursor to 0 in that case and as that sets the cursor at the end, all following emojis are set behind it as expected.


Reading through this now, I am not sure this is 100% compatible with @gtalha07 latest changes where we might have set the text code even if it never had focus. Then we'd add the icons at the beginning, though the expectation would probably be to have it at the end. thus, I think, I'll change this code again ...